### PR TITLE
Add SEQUENCE NAME to postgres sequence options

### LINKED ISSUE #1857: Postgres SEQUENCE NAME property not parsed
## Expected Behaviour
SQLFluff should parse SQL files generated by `pg_dump`

## Observed Behaviour
Postgres dump utility produces undocumented SQL code for identity sequences. It creates a `SEQUENCE NAME identifier` property, which I was not able to find in the official documentation. SQLFluff fails to parse this construct.

## Steps to Reproduce
Create a file `test.sql` with the following contents (originally produced by `pg_dump`):
```
ALTER TABLE public.history ALTER COLUMN id ADD GENERATED ALWAYS AS IDENTITY (
    SEQUENCE NAME public.history_id_seq
    START WITH 1
    INCREMENT BY 1
    NO MINVALUE
    NO MAXVALUE
    CACHE 1
);
```
`psql` correctly handles this code, while SQLFluff fails:
```
markpolyak$ sqlfluff lint --dialect=postgres test.sql
== [test.sql] FAIL
L:   1 | P:  76 |  PRS | Line 1, Position 76: Found unparsable section: ' (\n
                       | SEQUENCE NAME public.history_id_s...'
All Finished 📜 🎉!
```
If the `SEQUENCE NAME ...` line is commented out SQLFluff is able to parse that file without errors.

## Dialect
postgres

## Version
sqlfluff, version 0.8.1
Python 3.7.6

## Configuration
default

### PR DESCRIPTION
**Pull request by jpy-git**
### Brief summary of the change made

This PR is intended to fix #1857. Adds `SEQUENCE NAME <sequence_identifier>` to the sequence options in the postgres dialect.

### Are there any other side effects of this change that we should be aware of?
Not that i'm aware of.

### Pull Request checklist
- [X] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `python test/generate_parse_fixture_yml.py` or by running `tox` locally).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

**Review by barrywhart — APPROVED** [review]
Looks good to me! Thanks.

Would you like me to merge it?

File: src/sqlfluff/dialects/dialect_postgres.py
Comment (WittierDinosaur) [inline src/sqlfluff/dialects/dialect_postgres.py:2044]: You mention in the issue that this isn't in the docs - for future readers, can you add an inline comment above this mentioning that this isn't documented?

Comment (jpy-git) [reply src/sqlfluff/dialects/dialect_postgres.py:2044]: @WittierDinosaur on it now :)

Comment (jpy-git) [reply src/sqlfluff/dialects/dialect_postgres.py:2046]: good shout as it will stop future PRs from thinking it's a mistake and removing. I've added that comment in now.

**Comment by jpy-git** [comment]
> Looks good to me! Thanks.
> 
> Would you like me to merge it?

@barrywhart Thanks! I can't merge it as it says `Only those with write access to this repository can merge pull requests.` so happy for you to merge if needs be :)

**Comment by barrywhart** [comment]
I merged it. 👍🏽

**Pull request merged**
